elastic/kibana
# [maps] ES|QL adhoc data view

Fixes https://github.com/elastic/kibana/issues/176325

https://github.com/elastic/kibana/pull/174736 resolved the issue of [Filter bar displays repeated same dataview names equal to the number of ES|QL panels based on that dataview in the dashboard](https://github.com/elastic/kibana/issues/168131). This PR uses `getESQLAdHocDataview` to create an adhoc data view for ESQL statements.

### Test steps
1. install sample web logs data set. Wait for install to finish (so web logs is default data view)
2. install sample flights data set.
3. create new map
4. add ESQL layer. Set statement to `from kibana_sample_data_flights | keep DestLocation | limit 10000`. Click run.
5. Verify data view is passed to unified search and flights fields are displayed in typeahead in unified search
6. save map
7. open map in new tab. Verify adhoc data view is re-hydrated in new tab and unified search works as expected
